syntax = "proto2";

package ostorlab.runtimes.proto;


message Arg {
  // Defines the arguments passed to the agent. All arg values are serialized to raw bytes and define a type that the
  // must know how to handle.
  required string name = 1;
  required string type = 2;
  required bytes  value = 3;
}

message PortMapping {
  // Defines container port mapping. All ports are mapping on 0.0.0.0.
  required sint32 source_port = 1;
  required sint32 destination_port = 2;
}


message AgentInstanceSettings {
  // Proto definition of agent instance settings. This is use to serialize all the settings passed from the runtime to
  // the agent.
  required string key = 1;
  required string bus_url = 2;
  required string bus_exchange_topic = 3;
  optional string bus_managment_url = 4;
  optional string bus_vhost = 5;
  repeated Arg args = 6;
  repeated string constraints = 7;
  repeated string mounts = 8;
  required string restart_policy = 9;
  optional sint64 mem_limit = 10;
  repeated PortMapping open_ports = 11;
  required sint32 replicas = 12;
  required string healthcheck_host = 13;
  required sint32 healthcheck_port = 14;
}
